From f7176c8223bcec1e375998ca1f4cf788594062ff Mon Sep 17 00:00:00 2001
From: Sasha Smundak <asmundak@google.com>
Date: Fri, 17 Aug 2018 12:41:24 -0700
Subject: [PATCH 4/4] Do not call sort when setting ALL_DEPS.MODULES.

This particular invocation of 'sort' is expensive (it may amount to 30%
of the total ckati execution time for the clean build).

Test: verify that the result of running 'm nothing' is unchanged,
verify that the result of running 'm -j deps-license
PROJ_PATH=packages/app/% DEP_PATH=external/%' is unchanged.

Change-Id: If1cfddd4dee24559a26ecceebdf03cf49cc5a367
---
 core/base_rules.mk          | 2 +-
 core/tasks/deps_licenses.mk | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/base_rules.mk b/core/base_rules.mk
index 046d911b7..51f8b5cd0 100644
--- a/core/base_rules.mk
+++ b/core/base_rules.mk
@@ -694,7 +694,7 @@ INSTALLABLE_FILES.$(LOCAL_INSTALLED_MODULE).MODULE := $(my_register_name)
 ##########################################################
 # Track module-level dependencies.
 # Use $(LOCAL_MODULE) instead of $(my_register_name) to ignore module's bitness.
-ALL_DEPS.MODULES := $(sort $(ALL_DEPS.MODULES) $(LOCAL_MODULE))
+ALL_DEPS.MODULES := $(ALL_DEPS.MODULES) $(LOCAL_MODULE)
 ALL_DEPS.$(LOCAL_MODULE).ALL_DEPS := $(sort \
   $(ALL_MODULES.$(LOCAL_MODULE).ALL_DEPS) \
   $(LOCAL_STATIC_LIBRARIES) \
diff --git a/core/tasks/deps_licenses.mk b/core/tasks/deps_licenses.mk
index bb20fa042..daf986f60 100644
--- a/core/tasks/deps_licenses.mk
+++ b/core/tasks/deps_licenses.mk
@@ -40,7 +40,7 @@ $(if $(_gmad_new),$(eval $(1) += $(_gmad_new))\
 endef
 
 define print-deps-license
-$(foreach m, $(ALL_DEPS.MODULES),\
+$(foreach m, $(sort $(ALL_DEPS.MODULES)),\
   $(eval m_p := $(sort $(ALL_MODULES.$(m).PATH) $(ALL_MODULES.$(m)$(TARGET_2ND_ARCH_MODULE_SUFFIX).PATH)))\
   $(if $(filter $(PROJ_PATH),$(m_p)),\
     $(eval deps :=)\
-- 
2.17.1

